tests/marqo_test.py

```diff
@@ -146,4 +142,39 @@ def warm_request(self, func, *args, **kwargs):
         This solves the occurence of tests failing due to eventual consistency implemented in marqo cloud.
         '''
         for i in range(5):
-            func(*args, **kwargs) 
+            func(*args, **kwargs)
+
+    def create_cloud_index(self, index_name, index_defaults=None, **kwargs):
+        client = marqo.Client(**self.client_settings)
+        index_settings = index_defaults if index_defaults else {}
+        index_settings.update({
+            "inference_type": "marqo.CPU", "storage_class": "marqo.basic", "model": "hf/all_datasets_v4_MiniLM-L6"
+        })
+        while True:
+            if client.http.get(f"/indexes/{index_name}/status") == "READY":
+                break
+            try:
+                client.create_index(index_name, settings_dict=index_settings)
+            except marqo.errors.MarqoWebError as e:
+                if e.status_code == 409:
+                    continue
+
+    def create_index(self, index_name, index_defaults=None, **kwargs):
+        client = marqo.Client(**self.client_settings)
+        if client.config.is_marqo_cloud:
+            self.create_cloud_index(index_name, index_defaults, **kwargs)
+            return
+        client.create_index(index_name, settings_dict=index_defaults, **kwargs)
+
+    def delete_documents(self, index_name):
+        client = marqo.Client(**self.client_settings)
+        indexes = client.get_indexes()
+        if index_name in indexes:
+            if client.http.get(f"/indexes/{index_name}/status") == "READY":
+                try:
+                    idx = client.index(index_name)
+                    docs_to_delete = [i['_id'] for i in idx.search("")['hits']]
```

Comment:
we need to do this in a for loop probably if number of docs in index > limit. We can check if index.get_stats()["numbeOfDocuments"] == 0, and then break